libtenzir/include/tenzir/cast.hpp

```diff
@@ -261,6 +293,45 @@ struct cast_helper<list_type, list_type> {
              from_array->null_bitmap(), from_array->null_count())
       .ValueOrDie();
   }
+
+  static auto cast_value(const list_type& from_type, const list& value,
+                         const list_type& to_type) noexcept
+    -> caf::expected<type_to_data_t<list_type>> {
+    if (from_type == to_type)
+      return value;
+    auto output = to_type.construct();
```

Comment:
```suggestion
    output.reserve(value.size());
```

Reply:
done in 75dbddf6033f4098ed572de5ec48384401897aef